core/src/main/kotlin/com/malinskiy/marathon/execution/Scheduler.kt

```diff
@@ -18,22 +22,22 @@ import kotlinx.coroutines.experimental.launch
  * 5. Batching:     TestBatch into manageable chunks
  * 6. Retries:      Retry if something fails and we didn't account for it in the flakiness
  */
+private const val DEFAULT_INITIAL_DELAY_MILLIS = 10_000L
 
 class Scheduler(private val deviceProvider: DeviceProvider,
                 private val analytics: Analytics,
                 private val configuration: Configuration,
-                private val list: Collection<Test>) {
+                private val tests: Collection<Test>) {
 
-    companion object {
-        private const val DEFAULT_INITIAL_DELAY_MILLIS = 10_000L
-    }
-
-    private val pools = mutableMapOf<DevicePoolId, Aktor<DevicePoolMessage>>()
+    private val pools = mutableMapOf<DevicePoolId, SendChannel<FromScheduler>>()
     private val poolingStrategy = configuration.poolingStrategy
 
+    private val logger = KotlinLogging.logger("Scheduler")
+
     suspend fun execute() {
         subscribeOnDevices()
-        healthCheck(startDelay = DEFAULT_INITIAL_DELAY_MILLIS) {
+        waitWhileTrue(startDelay = DEFAULT_INITIAL_DELAY_MILLIS) {
```

Comment:
why we need this delay? (comment)

Reply:
We wait 10 sec for first connected device, because ADB provides devices with small delay

Reply:
i think we must comment such things in code, it's not so intuitive and hard to test properly